packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_SaPFactory_def.hpp

```diff
@@ -148,6 +148,14 @@ namespace MueLu {
     // By default, we don't need global constants for SaP
     APparams->set("compute global constants: temporaries",APparams->get("compute global constants: temporaries",false));
     APparams->set("compute global constants",APparams->get("compute global constants",false));
+// *** This section is dubious CBL
+    auto &APpsub = APparams->sublist( "matrixmatrix: kernel params",false);
+    bool isMM  = APparams->get("isMatrixMatrix_TransferAndFillComplete",false);
+    int  mmOCC = APparams->get("MM_TAFC_OptimizationCoreCount",2997);
```

Comment:
What is special about the number 2977?

Reply:
Also, that guy is 3000 in other places.  Should we set that centrally somewhere in Tpetra?

Reply:
> Also, that guy is 3000 in other places. Should we set that centrally somewhere in Tpetra?

It seems weird to me to have a default value set here.  But maybe that's because this is special to A*P, where P is tall and skinny, and MueLu is the only place this is important?   Anyway, I like @csiefer2's suggestion of setting this somewhere in Tpetra instead.

Reply:
I think that's just the empirically determined crossover point for where this algorithm pays off.